### DISCUSSION TITLE
How to detect if a user is loggedin?

### DISCUSSION THREAD

**Discussion by margenn** [adminerevo/adminerevo#211] [Q&A] [answered]
I'm improving a plugin that transform ordinary fields in a dropdown based on values from another table.

The problem is: when session is over by timeout, the url points to edit form, but the gui presents the login form.

The plugin only checked the url, "thinks" that everything is OK and continue the process, loading the javascript and so on.

I've found that a specific $_SESSION variable becames null when user is logged OUT, and I'm using it.
It works, but maybe someone knows a better way of doing this.

```php
public function head() {
	$editInterface = isset($_GET['edit']);
	$userLoggedin = $_GET['username'] && $_SESSION['dbs']['server'][''][$_GET['username']];
	if (! ($editInterface && $userLoggedin) ) {
		return; // quit
	}
	// All bellow only if user is loggedin in the edit screen
```

**Comment by margenn** [comment]
The previous detection method fails in the first page right after login when a row-edit-page is called from browser favorites. 

A more reliable method is checking if the global variable "$fields" has values. The problem in this method, is that it was tested only with edit interface, and probably works only in this interface. It isn't an universal solution.

```php
// global $fields can't be null
global $fields;
if (! $fields ) { return; }
```

Anyway, the plugin improvements are done: 
https://github.com/margenn/adminer-foreign-datalist/blob/main/foreign-datalist.php

**Reply by mckaygerhard** [reply]
wow.. thanks for your contribution notes.

in your profile there are several others plugins.. interesting.. and seems you are linux users (with winbuntu but linux at least!)

**Reply by margenn** [reply]
`global $fields;` does'n work with compiled adminer, so it has to be changed to `$fields = fields($_GET["edit"]);`
The same with `global $adminer;` that became `$adminer = adminer();`

**Comment by margenn** [comment, marked as answer]
This seems to work (at least for edit), so closing this thread.

```php
// Check if user is online on edit interface
$fields = fields($_GET["edit"]);
if (! $fields ) { return; }
```
